apps/crn-server/scripts/export-analytics.ts

```diff
@@ -32,27 +33,44 @@ export const exportAnalyticsData = async (
   await file.write('[\n');
   metric === 'team-leadership'
     ? await exportData(metric, file)
-    : await exportDataForRanges(metric, file);
+    : await exportDataWithFilters(metric, file);
 
   await file.write(']');
 };
 
-const exportDataForRanges = async (
+const exportDataWithFilters = async (
   metric: Metric,
   file: FileHandle,
 ): Promise<void> => {
-  for (let i = 0; i < timeRanges.length; i += 1) {
-    await exportData(metric, file, timeRanges[i]);
-    if (i != timeRanges.length - 1) {
-      await file.write(',');
+  if (metric === 'user-productivity') {
+    for (let i = 0; i < timeRanges.length; i += 1) {
+      for (let j = 0; j < documentCategories.length; j += 1) {
+        await exportData(metric, file, {
+          timeRange: timeRanges[i],
+          documentCategory: documentCategories[j],
+        });
+        if (j != documentCategories.length - 1) {
+          await file.write(',');
+        }
+      }
+      if (i != timeRanges.length - 1) {
```

Comment:
looks to me like you could get double `,` if both `j` and `i` reach the lengths of their corresponding arrays, not that it should matter (it would just be seen as an empty row but im sure algolia wouldn't index an empty row)

This is quite a complicated logic, would be nice to test this file at some point (not now)

Reply:
I don't think that's the case as we don't include the `,` when the `j` and `i` reach the end of their arrays? That is what the condition above is for. I do agree about testing the file at some point